types/lodash/fp/index.d.ts

```diff
@@ -0,0 +1,618 @@
+// Created by https://github.com/albohlabs
+// Pull request by https://github.com/thalesmello
+// Originally taken from https://gist.github.com/albohlabs/1c864d83b17e4e99edff210729327e03
+
+declare namespace fp {
+
+    interface Dictionary<T> {
+        [index: string]: T;
+    }
+
+    interface CurriedFunction1<T1, R> {
+        (): CurriedFunction1<T1, R>;
+        (t1: T1): R;
+    }
+
+    interface CurriedFunction2<T1, T2, R> {
+        (): CurriedFunction2<T1, T2, R>;
+        (t1: T1): CurriedFunction1<T2, R>;
+        (t1: T1, t2: T2): R;
+    }
+
+    interface CurriedFunction3<T1, T2, T3, R> {
+        (): CurriedFunction3<T1, T2, T3, R>;
+        (t1: T1): CurriedFunction2<T2, T3, R>;
+        (t1: T1, t2: T2): CurriedFunction1<T3, R>;
+        (t1: T1, t2: T2, t3: T3): R;
+    }
+
+    interface CurriedFunction4<T1, T2, T3, T4, R> {
+        (): CurriedFunction4<T1, T2, T3, T4, R>;
+        (t1: T1): CurriedFunction3<T2, T3, T4, R>;
+        (t1: T1, t2: T2): CurriedFunction2<T3, T4, R>;
+        (t1: T1, t2: T2, t3: T3): CurriedFunction1<T4, R>;
+        (t1: T1, t2: T2, t3: T3, t4: T4): R;
+    }
+
+    interface CurriedFunction5<T1, T2, T3, T4, T5, R> {
+        (): CurriedFunction5<T1, T2, T3, T4, T5, R>;
+        (t1: T1): CurriedFunction4<T2, T3, T4, T5, R>;
+        (t1: T1, t2: T2): CurriedFunction3<T3, T4, T5, R>;
+        (t1: T1, t2: T2, t3: T3): CurriedFunction2<T4, T5, R>;
+        (t1: T1, t2: T2, t3: T3, t4: T4): CurriedFunction1<T5, R>;
+        (t1: T1, t2: T2, t3: T3, t4: T4, t5: T5): R;
+    }
+
+    interface ListIterator<T, TResult> {
+        (value: T, index: number, collection: List<T>): TResult;
+    }
+
+    // Common interface between Arrays and jQuery objects
+    interface List<T> {
+        [index: number]: T;
+        length: number;
+    }
+
+    interface Static {
+
+        /**
+         * Creates an array of elements corresponding to the given keys, or indexes, of collection. Keys may be
+         * specified as individual arguments or as arrays of keys.
+         *
+         * @param props The property names or indexes of elements to pick, specified individually or in arrays.
+         * @param collection The collection to iterate over.
+         * @return Returns the new array of picked elements.
+         */
+        at<T>(props: (number | string | (number | string)[])[]): (collection: List<T>) => T[];
+
+        /**
+         * Returns a curried equivalent of the provided function, with the specified arity. The curried function has
+         * two unusual capabilities. First, its arguments needn't be provided one at a time.
+         */
+        curryN(length: number, fn: (...args: any[]) => any): Function;
```

Comment:
Never use `Function`. Instead use a more specific function type. In this case, probably `(...args: any[]) => any`